### PR TITLE
Monorepo docs

### PR DESCRIPTION
**Pull request by silkeh**
## Description

- Update the docs for working with a monorepo.
- Replace `make` by `go-task`.

Depends on #373

### Submitter Checklist

- [x] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: docs/packaging/git-basics.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## Working with branches
+
+The recommended way to switch branches is `git switch`.
+To create a new branch use `git switch -c`.
+
+## Keeping your branch up to date with the master branch
+
+If there any new changes to the repository whilst you are still working on your branch you need to [sync your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork).
+Run `git fetch`, and `git rebase origin/master` to pull the changes in.
```
Comment (Staudey) [inline docs/packaging/git-basics.md:16]:
```suggestion
Run `git fetch`, and `git rebase origin/main` to pull the changes in.
```

File: docs/packaging/git-basics.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+You will need to fix any conflicts manually.
+You can check your branch against master with `git diff origin/master..my-branch`.
```
Comment (Staudey) [inline docs/packaging/git-basics.md:18]:
```suggestion
You can check your branch against main with `git diff origin/main..my-branch`.
```

File: docs/packaging/git-basics.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Now that you've tested and reviewed your change, when you're happy with it, it's time to commit your changes with `git commit`.
+
+## Deleting your branch after a pull request is merged
+
+Once your pull request has been accepted you can delete your working branches.
+First, change to the package directory, switch to the master branch, then delete your local and the remote branches.
+
+```bash
+cd /home/solus-builds/package
+git switch master
```
Comment (Staudey) [inline docs/packaging/git-basics.md:54]:
```suggestion
git switch main
```

File: docs/packaging/creating-a-new-package.md [hunk 4]
```diff
@@ -1,37 +1,94 @@
...
+
+## Fork the getsolus/packages Repository / Update Your Fork
+
+### Create a Fork
+
+If you had not yet done so already, fork the repo [getsolus/packages](https://github.com/getsolus/packages) using the GitHub web UI or [`gh` cli tool](https://cli.github.com/manual/gh_repo_fork) from the `github-cli` package. It will be forked to `github.com/yourgithubaccount/packages`.
+
+### Update a Fork
+
+If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the master branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```
Comment (Staudey) [inline docs/packaging/creating-a-new-package.md:29]:
```suggestion
If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the main branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```

File: docs/packaging/creating-a-new-package.md [hunk 5]
```diff
@@ -1,37 +1,94 @@
...
+cd /home/solus-builds/packages
 ```
 
+### Update an existing clone
+
+If you already have a local clone, you need to bring it up to date. To do this run:
+
+```bash
+cd /home/solus-builds/packages
+git switch master
```
Comment (Staudey) [inline docs/packaging/creating-a-new-package.md:49]:
```suggestion
git switch main
```

File: docs/packaging/git-basics.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+# Git Basics
+
+## Working with branches
+
+The recommended way to switch branches is `git switch`.
+To create a new branch use `git switch -c`.
+
+## Keeping your branch up to date with the master branch
```
Comment (Staudey) [inline docs/packaging/git-basics.md:13]:
```suggestion
## Keeping your branch up to date with the main branch
```

File: docs/packaging/git-basics.md (hunk 1 above)
Comment (Staudey) [inline docs/packaging/git-basics.md:16]:
```suggestion
Run `git fetch`, and `git rebase origin/main` to pull the changes in.
```

File: docs/packaging/git-basics.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+**Check your branch**
+
+Get the status of the branch with `git status`. Make sure all the files you changed are staged, and that there are no untracked files. The git status should say your branch is clean.
+
+Now that you've tested and reviewed your change, when you're happy with it, it's time to commit your changes with `git commit`.
+
+## Deleting your branch after a pull request is merged
+
+Once your pull request has been accepted you can delete your working branches.
+First, change to the package directory, switch to the master branch, then delete your local and the remote branches.
```
Comment (Staudey) [inline docs/packaging/git-basics.md:50]:
```suggestion
First, change to the package directory, switch to the main branch, then delete your local and the remote branches.
```

File: docs/packaging/updating-an-existing-package.md [hunk 8]
```diff
@@ -1,23 +1,81 @@
...
+
+## Fork the getsolus/packages Repository / Update Your Fork
+
+### Create a Fork
+
+If you had not yet done so already, fork the repo [getsolus/packages](https://github.com/getsolus/packages) using the GitHub web UI or [`gh` cli tool](https://cli.github.com/manual/gh_repo_fork) from the `github-cli` package. It will be forked to `github.com/yourgithubaccount/packages`.
+
+### Update a Fork
+
+If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the master branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```
Comment (Staudey) [inline docs/packaging/updating-an-existing-package.md:31]:
```suggestion
If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the main branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```

**Review by Staudey — COMMENTED** [review]
The default branch of getsolus/packages is now `main`, unless that is supposed to change

**Comment by silkeh** [comment]
@Staudey: good catch. Thanks!

File: docs/packaging/prepare-for-packaging.mdx
Comment (TraceyC77) [inline docs/packaging/prepare-for-packaging.mdx:1]: The "Setting up a Build Directory" section needs to be updated. If you follow it as is you will wind up with two folders for `common`

```
~/solus-builds/common
~/solus-builds/packages/common
```

The symlink commands also need to be updated to point to `~/solus-builds/packages/common`
The packaging folder layout also needs to be updated.

File: docs/packaging/updating-an-existing-package.md [hunk 9]
```diff
@@ -37,8 +38,8 @@ If you do not yet have a clone of the package, change to your packaging director
```
Comment (TraceyC77) [inline docs/packaging/updating-an-existing-package.md:38]: On the previous line 
"If you do not yet have a clone of the package..."

should be
"If you do not yet have a clone of your packages repo fork, change to your packaging directory and clone your fork. Then, switch to the  directory of the package to update. For example:"

File: docs/packaging/updating-an-existing-package.md [hunk 10]
```diff
@@ -47,7 +48,7 @@ If you already have a local clone, you need to bring it up to date. To do so run
 
 ```bash
 cd ~/solus-builds/package-name
```
Comment (TraceyC77) [inline docs/packaging/updating-an-existing-package.md:50]: This needs to be udpated

```suggestion
cd ~/solus-builds/packages/packages/n/nano
```

**Review by TraceyC77 — CHANGES_REQUESTED** [review]
Just a few updates needed, then this is good to go. Good work overall.

File: docs/packaging/creating-a-new-package.md [hunk 11]
```diff
@@ -26,7 +26,7 @@ If you had not yet done so already, fork the repo [getsolus/packages](https://gi
 
 ### Update a Fork
 
-If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the master branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
+If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the main branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```
Comment (EbonJaeger) [inline docs/packaging/creating-a-new-package.md:29]: Thinking the name of the branch should be made clear that it isn't just a word in the sentence.
```suggestion
If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the `main` branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```

File: docs/packaging/updating-an-existing-package.md [hunk 12]
```diff
@@ -19,15 +19,16 @@ This PR resolves software update request https://github.com/getsolus/packages/is
...
+If you had not yet done so already, fork [getsolus/packages](https://github.com/getsolus/packages) using the GitHub web UI or [`gh` cli tool](https://cli.github.com/manual/gh_repo_fork) from the `github-cli` package. It will be forked to `github.com/yourgithubaccount/packages`.
 
 ### Update a Fork
 
-If you already have a fork of the package in GitHub, from previous updates, log into GitHub. Make sure you're looking at the master branch. Check to see that your fork is up to date with the repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
+If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the main branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```
Comment (EbonJaeger) [inline docs/packaging/updating-an-existing-package.md:31]: Same as above.
```suggestion
If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the `main` branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```

**Review by EbonJaeger — CHANGES_REQUESTED** [review]
Very good, overall!

File: docs/packaging/prepare-for-packaging.mdx [hunk 13]
```diff
@@ -40,26 +40,26 @@ The main packages we need are provided by the `system.devel` component.
 Namely:
 
 - `ypkg` the program that actually builds packages
-- `make` used by our build tools for scripting
 
 ```bash
```
[the remark this answers is not in the document]
Comment (silkeh) [reply docs/packaging/prepare-for-packaging.mdx]: I removed the `system.devel` component entirely, and just added `ypkg` separately.

File: docs/packaging/prepare-for-packaging.mdx [hunk 14]
```diff
@@ -40,26 +40,26 @@ The main packages we need are provided by the `system.devel` component.
 Namely:
```
[the remark this answers is not in the document]
Comment (silkeh) [reply docs/packaging/prepare-for-packaging.mdx]: I merged the two lists, so it should be clearer now.

**Review by EbonJaeger — APPROVED** [review]
LGTM, thanks! :)

**Review by TraceyC77 — APPROVED** [review]
LGTM. Great work.

**Pull request merged**